packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/internal/KtorNetworkTransport.kt

```diff
@@ -0,0 +1,199 @@
+                        customHeaders.forEach {
+                            append(it.key, it.value)
+                        }
+
+                        // 2. Then replace default authorization header with custom one if present
+                        headers
+//                            .filter {
+//                                // Ignore Content-Type header as it's handled by the engine
+//                                it.key != HttpHeaders.ContentType
+//                            }
+                            .toMutableMap().also { receivedHeaders ->
+                                val authorizationHeaderValue =
+                                    headers[DEFAULT_AUTHORIZATION_HEADER_NAME]
+
+                                if (authorizationHeaderValue != null &&
+                                    authorizationHeaderName != DEFAULT_AUTHORIZATION_HEADER_NAME
+                                ) {
+                                    receivedHeaders.remove(DEFAULT_AUTHORIZATION_HEADER_NAME)
+                                    receivedHeaders[authorizationHeaderName] =
+                                        authorizationHeaderValue
+                                }
+
+                                // 3. Finally add all headers defined by Object Store
+                                receivedHeaders.forEach {
+                                    if (method != "get"
+                                        || !url.contains("profile")
+                                        || it.key != HttpHeaders.ContentType
+                                    ) {
+                                        append(it.key, it.value)
+                                    }
+                                }
+                            }
+                    }
+
+                    addBody(method, body)
+                    addMethod(method)
+                }
+
+                try {
+                    when (method) {
+                        "delete" -> client.delete<HttpResponse>(url, requestBuilderBlock)
+                        "patch" -> client.patch<HttpResponse>(url, requestBuilderBlock)
+                        "post" -> client.post<HttpResponse>(url, requestBuilderBlock)
+                        "put" -> client.put<HttpResponse>(url, requestBuilderBlock)
+                        "get" -> client.get<HttpResponse>(url, requestBuilderBlock)
+                        else -> throw IllegalArgumentException("Wrong request method: '$method'")
+                    }.let {
+                        processHttpResponse(it)
+                    }
+                } catch (e: ServerResponseException) {
+                    // 500s are thrown as ServerResponseException
+                    processHttpResponse(e.response)
+                } catch (e: CancellationException) {
+                    println("EXCEPTION 1: $e")
+                    println("MESSAGE 1: ${e.message}")
+                    Response(-1, -1, mapOf(), "")
```

Comment:
In Java we use `0` instead of `-1` to indicate not in use, we should probably do the same here.

Reply:
Translated the various exceptions into error response codes according to Java.